src/ngraph/op/fused/mod.cpp

```diff
@@ -0,0 +1,59 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/op/fused/mod.hpp"
+#include "ngraph/builder/make_constant.hpp"
+#include "ngraph/op/abs.hpp"
+#include "ngraph/op/convert.hpp"
+#include "ngraph/op/divide.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/sign.hpp"
+#include "ngraph/op/subtract.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::Mod::type_info;
+
+op::Mod::Mod(const Output<Node>& A, const Output<Node>& B, const AutoBroadcastSpec& auto_broadcast)
+    : FusedOp({A, B})
+    , m_auto_broadcast(auto_broadcast)
+{
+}
+
+NodeVector op::Mod::decompose_op() const
+{
+    const auto dividend = make_shared<op::Abs>(input_value(0));
+    const auto dividend_sign = make_shared<op::Sign>(input_value(0));
+    const auto dividend_et = dividend->get_element_type();
+    const auto divisor = make_shared<op::Abs>(input_value(1));
+
+    // truncated(a / b)
+    auto division = make_shared<op::Convert>(
+        make_shared<op::Divide>(dividend, divisor, m_auto_broadcast), ngraph::element::i64);
+    division = make_shared<op::Convert>(division, dividend_et);
+    // truncated(a / b) * b
+    const auto multiplication = make_shared<op::Multiply>(division, divisor, m_auto_broadcast);
+    // a mod b = a - truncated(a / b) * b
+    const auto mod = make_shared<op::Subtract>(dividend, multiplication, m_auto_broadcast);
+
+    // apply sign of dividend
+    return {make_shared<op::Multiply>(dividend_sign, mod, m_auto_broadcast)};
```

Comment:
Can we use V1 ops here?

Reply:
Yes, I changed used ops to `v1` (if `v1` were provided)